### PR TITLE
Merge hotfix 3.3.3 to release-4.0.0

### PR DESCRIPTION
**Pull request by adamralph**
Connects to #175

### CODE REVIEW

File: src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs [hunk 1]
```diff
@@ -30,7 +30,8 @@ public ConnectionFactory(ConnectionConfiguration connectionConfiguration)
                 Password = connectionConfiguration.Password,
                 RequestedHeartbeat = connectionConfiguration.RequestedHeartbeat,
                 AutomaticRecoveryEnabled = true,
-                NetworkRecoveryInterval = connectionConfiguration.RetryDelay
+                NetworkRecoveryInterval = connectionConfiguration.RetryDelay,
+                UseBackgroundThreadsForIO = true,
```
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs]: @bording I'm not sure if we needed to this or not in the 4.0.0 transport.

Was this part of fixing the hanging tests in 3.x or is it something else?

File: src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs [hunk 2]
```diff
@@ -55,7 +56,6 @@ public IConnection CreateConnection(string connectionName)
         {
             lock (lockObject)
             {
-                connectionFactory.ClientProperties["purpose"] = connectionName;
```
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs]: @bording I presume this is also appropriate in the 4.0 transport after upgrading the client to 3.6.3?

**Comment by adamralph** [comment]
I double checked all the disposals in the 4.0 transport (wrt to https://github.com/Particular/NServiceBus.RabbitMQ/pull/176/commits/f7ab916958dcc5980ea463b8b2dcc55ffaf4c136) and, as I suspected, everything is OK.

File: src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs (hunk 2 above)
Comment (bording) [inline src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs]: That's correct. Setting the purpose value is redundant now that the connection name property is displayed properly in the RabbitMQ management UI.

File: src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs (hunk 1 above)
Comment (bording) [inline src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs]: I did end up making that change in relation to working around the problem with the V5 acceptance test framework, but I think it's an OK change to make regardless. It's actually rather odd that the connection threads aren't background threads by default anyway.

Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs]: I'm happy with the change then.

**Comment by adamralph** [comment]
Removed WIP. Happy for it to be merged.

**Comment by bording** [comment]
Just noticed, looks like you missed the nuspec version bump.

**Comment by adamralph** [comment]
Ah, well spotted! Will fix.

**Comment by adamralph** [comment]
@bording I amended the commit to add the nuspec bump

**Comment by adamralph** [comment]
@bording the offending comma is gone :smiley:

**Pull request merged**